src/connectors/h1_collect.py

```diff
@@ -0,0 +1,251 @@
+"""HackerOne Data
+Collect HackerOne reports and payment transactions using an API key
+"""
+import os
+from datetime import datetime
+from email.utils import parsedate_to_datetime
+from urllib.error import HTTPError
+
+import fire
+import requests
+from runners.helpers import db, log
+from runners.helpers.dbconfig import ROLE as SA_ROLE
+
+from connectors.utils import yaml_dump
+
+PAGE_SIZE = 500
+
+CONNECTION_OPTIONS = [
+    {
+        'name': 'api_token',
+        'title': "API Token",
+        'prompt': "Available in your H1 Settings",
+        'type': 'str',
+        'secret': True,
+        'required': True,
+    },
+    {
+        'name': 'api_identifier',
+        'title': "API Identifier",
+        'prompt': "Available in your H1 Settings",
+        'type': 'str',
+        'secret': True,
+        'required': True,
+    },
+    {
+        'name': 'program_name',
+        'title': "Program Name",
+        'prompt': "Your handle in Program Settings",
+        'type': 'str',
+        'required': True,
+    },
+    {
+        'name': 'account_id',
+        'title': "Account ID",
+        'prompt': "https://api.hackerone.com/core-resources/#programs-get-your-programs",
+        'type': 'str',
+        'required': False,
+    },
+]
+
+LANDING_TABLE_COLUMNS_TRANSACTIONS = [
+    ('recorded_at', 'TIMESTAMP_LTZ(9)'),
+    ('type', 'string'),
+    ('activity_date', 'TIMESTAMP_LTZ(9)'),
+    ('activity_description', 'string'),
+    ('bounty_award', 'int'),
+    ('bounty_fee', 'int'),
+    ('debit_or_credit_amount', 'int'),
+    ('balance', 'int'),
+    ('id', 'number'),
+    ('url', 'string'),
+]
+
+LANDING_TABLE_COLUMNS_REPORTS = [
+    ('recorded_at', 'TIMESTAMP_LTZ(9)'),
+    ('id', 'number'),
+    ('type', 'string'),
+    ('title', 'string'),
+    ('state', 'string'),
+    ('created_at', 'TIMESTAMP_LTZ(9)'),
+    ('vulnerability_information', 'string'),
+    ('triaged_at', 'TIMESTAMP_LTZ(9)'),
+    ('closed_at', 'TIMESTAMP_LTZ(9)'),
+    ('first_program_activity_at', 'TIMESTAMP_LTZ(9)'),
+    ('bounty_awarded_at', 'TIMESTAMP_LTZ(9)'),
+    ('rating', 'string'),
+    ('name', 'string'),
+    ('description', 'string'),
+    ('external_id', 'string'),
+    ('asset_identifier', 'string'),
+    ('awarded_amount', 'int'),
+    ('awarded_bonus_amount', 'int'),
+]
+
+
+def load_data(url: str, token: str, api_identifier: str, params: dict = {}) -> dict:
+    '''Perform the API call'''
+    headers: dict = {"Accept": "application/json"}
+
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        resp = requests.get(
+            url, auth=(api_identifier, token), params=params, headers=headers
+        )
+        resp.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"HTTP error occurred: {http_err}")
+        raise
+    log.debug(resp.status_code)
+    return resp
```

Comment:
Merged in changes on v1.10.0 so that we can run the automated tests which [reported](https://app.circleci.com/pipelines/github/snowflakedb/SnowAlert/426/workflows/cb8b86ff-48be-47f9-b3cc-a8399b885c7f/jobs/4550):

    connectors/h1_collect.py:100: error: Incompatible return value type (got "Response", expected "Dict[Any, Any]")

Reply:
Fixed the issue. It was the return type hint. Changed that to `requests.Response`